### PR TITLE
1.8 Documentation updates

### LINKED ISSUE #11369: Clarify potential for connection disruption due to --native-routing-cidr flag
In Cilium 1.8, we intend to change the masquerade logic to more accurately determine when to apply masquerade (ie SNAT) to traffic in direct-routing mode, to ensure that traffic is only masqueraded when the destination is not directly reachable. To do this, we solicit the directly routable CIDR range from the user via the `--native-routing-cidr` option.

The upgrade docs already describe this briefly:
https://github.com/cilium/cilium/blob/master/Documentation/install/upgrade.rst#important-changes-required-before-upgrading-to-180

By my understanding, if this option is not configured, this will result in behaviour change for direct routing users which may lead to connection disruption. We should briefly extend it to clearly define the consequences of failing to configure the option to ensure that users treat this option with the severity it requires.

Questions to guide this documentation:
* [ ] Are all existing direct-routing users already explicitly configuring `--ipv4-cluster-cidr-mask-size` or is it automatically configured? If it is automatically configured, users may not be aware that they were running with this option.
  * By my initial grep for this option in the Cilium-1.7 codebase, it looks like no instructions inform the user that they must specify this CIDR.
  * If it's autodetected today, then we should make this clear that users who have no knowledge of this setting, but are running `--tunnel=disabled` will *still* need to read the paragraph and take action. This can be achieved with a short sentence that states what the default behaviour is, eg "Cilium 1.7 and below inferred the setting for this value if it was not explicitly set."
* [ ] What are the consequences of failing to configure this option?
  * For example, cilium-managed pods may be unable to establish connections to the outside world(?), (or to non-Cilium-managed IPs within the cluster?).
* [ ] Why does the user need to consider this now?
  * For example, "Previous Cilium versions inferred the cluster range to choose when to perform masquerading, however in some cases it would mistakenly masquerade traffic that is destined outside the cluster".

### LINKED ISSUE #11868: helm global.eni=true does not set ipam: eni
This does not seem correct to me, currently the https://docs.cilium.io/en/latest/gettingstarted/aws-eni/ guide has the option `global.eni` set but the `ipam` is set to `"cluster-pool"`. Is this intended?

```
helm install cilium ./cilium \
  --namespace kube-system \
  --set global.eni=true \
  --set global.egressMasqueradeInterfaces=eth0 \
  --set global.tunnel=disabled \
  --set global.nodeinit.enabled=true
```

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #11868
Fixes: #11369

### CODE REVIEW

File: Documentation/concepts/networking.rst [hunk 1]
```diff
@@ -189,3 +183,46 @@ this "ingress" load-balancing capability, which achieve the same effect that
...
+  CIDR of the local node. If the pod IPs are routable across a wider network,
+  that network can be specified with the option: ``native-routing-cidr:
+  10.0.0.0/8`` in which case all destinations within that CIDR will **not** be
+  masqueraded.
+
+Setting the masquerading interface
+  The default behavior will masquerade all traffic leaving on a non-Cilium
+  network device. This typically leads to the correct behavior. In order to
+  limit the network interface on which masquerading should be performed, the
+  option ``egress-masquerade-interfaces: eth+`` can be used.
```
Comment (joestringer) [inline Documentation/concepts/networking.rst]: maybe worth noting here what the `eth+` actually means, and that the user should specify a prefix pattern that matches the devices they care about.

I don't know if many modern Linux distributions still name devices as `ethN` by default these days.

Comment (joestringer) [reply Documentation/concepts/networking.rst]: (I'm also not sure if we're respecting this option for bpf masquerade /cc @brb )

**Review by joestringer — CHANGES_REQUESTED** [review]
LGTM, one minor comment below.

Would be nice to get a second view from someone like @seanmwinn  / @jedsalazar .

File: Documentation/concepts/networking.rst [hunk 2]
```diff
@@ -189,3 +183,46 @@ this "ingress" load-balancing capability, which achieve the same effect that
 Cilium handles forwarding and security only for ''internal'' traffic between
 different services.
 
+.. _concepts_masquerading:
+
+Masquerading
+============
+
+As the IPv4 addresses used for pods are typically allocated from RFC1918
```
Comment (jedsalazar) [inline Documentation/concepts/networking.rst]: I think it reads cleaner if we remove the condition in the beginning of the statement 
s/As the//

File: Documentation/concepts/networking.rst (hunk 1 above)
Comment (jedsalazar) [reply Documentation/concepts/networking.rst]: systemd-based distros have eliminated the eth naming scheme and replaced it with the systemd.net-naming-scheme (`enN`-based naming). that's not very intuitive so agreed that we note what `eth+` references

**Review by christarazi — COMMENTED** [review]
LGTM, when I looked at https://github.com/cilium/cilium/issues/11868 yesterday, I noticed that we have two of the same Helm deploy instructions for AWS / EKS / ENI. See [EKS guide](https://docs.cilium.io/en/latest/gettingstarted/k8s-install-eks/?highlight=config.ipam#deploy-cilium) and [ENI guide](https://docs.cilium.io/en/latest/gettingstarted/aws-eni/#deploy-cilium). It'd be nice if we can consolidate these two Helm instruction sections into one, by putting the shared info into a separate file and referencing it in the two sections. This would help avoid an inconsistency cropping up like the above issue.

File: Documentation/concepts/networking.rst [hunk 3]
```diff
@@ -189,3 +183,46 @@ this "ingress" load-balancing capability, which achieve the same effect that
...
+  The default behavior will masquerade all traffic leaving on a non-Cilium
+  network device. This typically leads to the correct behavior. In order to
+  limit the network interface on which masquerading should be performed, the
+  option ``egress-masquerade-interfaces: eth+`` can be used.
+
+Implementation Modes
+--------------------
+
+eBPF-based
+  The eBPF-based implementation is the most efficient
```
Comment (brb) [inline Documentation/concepts/networking.rst:229]: Once this PR is emerged, I will extend this section to add more info about device selection.

File: Documentation/install/upgrade.rst [hunk 4]
```diff
@@ -328,22 +328,47 @@ IMPORTANT: Changes required before upgrading to 1.8.0
...
+
+  Running with ``--masquerade=false`` (all chaining configurations)
+    No action required.
+
+  Running in direct-routing configuration (``--tunnel=disabled``)
+    The behavior has changed: Previously, the destination address range
+    excluded from masquerading was defined by the options ``--ipv4-range`` and
+    ``--ipv4-cluster-cidr-mask-size``. When unspecified, this was set to the
+    value ``10.0.0.0/8``. You **must** set the ``--native-routing-cidr`` option
+    and set it to the CIDR for which masquerading should be omitted for. This is
```
Comment (tklauser) [inline Documentation/install/upgrade.rst]: Nit:
```suggestion
    and set it to the CIDR for which masquerading should be omitted. This is
```

**Pull request merged**